azmq/detail/config/thread.hpp

```diff
@@ -0,0 +1,26 @@
+/*
+    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
+
+    This file is part of azmq
+
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+*/
+#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
+#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
+
+#if !defined(AZMQ_HAS_STD_THREAD)
```

Comment:
I've done it already for actor related stuff as part of a commit I hope to
push this coming week (I also cover mutex and condition_variable as part of
that commit as well). I will push a config/mutex.hpp and
config/condition_variable.hpp as part of that change today. I don't think
condition_variable is used anywhere else, though mutex is, and there are
threads used in the tests, so help fixing those up is very welcome.

On Fri, Apr 3, 2015 at 6:46 PM, Andrey Upadyshev <notifications@github.com>
wrote:

> In azmq/detail/config/thread.hpp
> <https://github.com/zeromq/azmq/pull/71#discussion_r27763268>:
>
> > @@ -0,0 +1,26 @@
> > +/*
> > +    Copyright (c) 2013-2014 Contributors as noted in the AUTHORS file
> > +
> > +    This file is part of azmq
> > +
> > +    Distributed under the Boost Software License, Version 1.0. (See accompanying
> > +    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
> > +*/
> > +#ifndef AZMQ_DETAIL_CONFIG_THREAD_HPP_
> > +#define AZMQ_DETAIL_CONFIG_THREAD_HPP_
> > +
> > +#if !defined(AZMQ_HAS_STD_THREAD)
>
> I think it's reasonable limitation. Anyway, "backing up" to Boost for
> standalone ASIO may be added later if requested by users.
>
> I can take a task to switch threading related stuff: std will be used by
> default unless AZMQ_DISABLE_STD_THREAD is defined.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeromq/azmq/pull/71/files#r27763268>.
>